File: core/src/com/gmaniliapp/flappyhermes/state/MenuState.java
package com.gmaniliapp.flappyhermes.state;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.gmaniliapp.flappyhermes.FlappyHermes;

public class MenuState extends State {

    private Texture background;
    private Texture startButton;

    public MenuState(GameStateManager gameStateManager) {
        super(gameStateManager);

        orthographicCamera.setToOrtho(false, FlappyHermes.WIDTH / 2, FlappyHermes.HEIGHT / 2);

        background = new Texture("background.png");

        startButton = new Texture("start_button.png");
    }

    @Override
    public void update(float deltaTime) {
        handleInput();
    }

    @Override
    public void render(SpriteBatch spriteBatch) {
        spriteBatch.setProjectionMatrix(orthographicCamera.combined);

        spriteBatch.begin();

        spriteBatch.draw(background, 0, 0);

        spriteBatch.draw(startButton, orthographicCamera.position.x - startButton.getWidth() / 2,
                orthographicCamera.position.y);

        spriteBatch.end();
    }

    @Override
    public void dispose() {
        background.dispose();
        startButton.dispose();
    }

    @Override
    protected void handleInput() {
        if (Gdx.input.justTouched()) {
            gameStateManager.set(new PlayState(gameStateManager));
        }
    }
}

File: core/src/com/gmaniliapp/flappyhermes/state/MenuState.kt
package com.gmaniliapp.flappyhermes.state

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.gmaniliapp.flappyhermes.FlappyHermes

class MenuState(gameStateManager: GameStateManager) : State(gameStateManager) {

    private val background: Texture
    private val startButton: Texture

    override fun update(deltaTime: Float) {
        handleInput()
    }

    override fun render(spriteBatch: SpriteBatch) {
        spriteBatch.projectionMatrix = orthographicCamera.combined
        spriteBatch.begin()
        spriteBatch.draw(background, 0f, 0f)
        spriteBatch.draw(startButton, orthographicCamera.position.x - startButton.width / 2f,
                orthographicCamera.position.y)
        spriteBatch.end()
    }

    override fun dispose() {
        background.dispose()
        startButton.dispose()
    }

    override fun handleInput() {
        if (Gdx.input.justTouched()) {
            gameStateManager.set(PlayState(gameStateManager))
        }
    }

    init {
        orthographicCamera.setToOrtho(false, FlappyHermes.WIDTH / 2f, FlappyHermes.HEIGHT / 2f)
        background = Texture("background.png")
        startButton = Texture("start_button.png")
    }
}